test/functorch/test_control_flow.py

```diff
@@ -2599,6 +2599,24 @@ def wrapper(x):
         res = torch.vmap(wrapper)(a)
         self.assertEqual(res, a + 1)
 
+    def test_cond_trace_set__and_mutation_with_graph_break(self):
+        def f(a, tmp):
+            a_view = a.view(-1)
+            with torch.no_grad():
+                a.set_(tmp)
+                a_view.mul_(2)
+            return a + tmp
```

Comment:
@ydwu4 what happens if one does the set_ on an intermediate tensor? i.e. 
```
def f(a, tmp):
    a = a.clone()
    tmp = tmp.clone() 
with torch.no_grad():
                a.set_(tmp)
                a_view.mul_(2)
  return a + tmp
```

Reply:
Just did some checking. We're getting a  cannot mutate tensors with frozen storage during functionalization pass in cond. This should be expected? Or we want it to fail at dynamo tracing time?

Reply:
If we torch.compile the cond:
```
@torch.compile
def f():
   return torch.cond(inp.sum() > 0, f, f, (inp, tmp))
```
does it fail at trace time or at runtime? William's point was that this should fail at trace time

Reply:
are the failures happening at trace-time or runtime?

Reply:
backend = "aot_eager" or "inductor" fails at tracing time.